doc/webservices/extract_api.rst

```diff
@@ -0,0 +1,483 @@
+
+        {"lines": [
+            {
+                "description": string,
+                "quantity": float,
+                "unit_price": float,
+                "product": string,
+                "taxes_amount": float,
+                "taxes": [
+                            {
+                                "amount": float, 
+                                "type": "fixed"|"percent", 
+                                "price_include": bool
+                            },
+                            ...
+                        ],
+                "subtotal": float,
+                "total": float
+            },
+            ...
+        ]}
+
+Response
+^^^^^^^^
+
+* ``jsonrpc``:  Same as for :ref:`/parse <webservices/extract_api/invoice_parse>`.
+* ``id``:  Same as for :ref:`/parse <webservices/extract_api/invoice_parse>`.
+* ``result``: 
+    * ``status_code``: code indicating the status of the request. ``status_code`` is 0 in case of success and not 0 otherwise. See table for more details about possible ``status_code``.
+    * ``status_msg``: A string giving verbose details about the request status.
+
+=============  ==========================================================
+ status_code       status_msg
+=============  ==========================================================
+0              Success
+12             Validation format is incorrect
+=============  ==========================================================
+
+The ``status_code`` 12 is only used in :ref:`integration mode <webservices/extract_api/integration_testing>`
+
+.. rst-class:: setup doc-aside
+
+.. switcher::
+
+    .. code-block:: json
+
+        {
+            "jsonrpc": string,
+            "id": string,
+            "result": {
+                "status_code": int,
+                "status_msg": string,
+            }
+        }
+
+
+.. _webservices/extract_api/integration_testing:
+
+Integration Testing 
+===================
+
+You can test your integration by using *integration_token* as ``account_token`` in the :ref:`/parse <webservices/extract_api/invoice_parse>` request. This token allows you to simulate the entire flow without really parsing a document.
+
+When calling the :ref:`/parse <webservices/extract_api/invoice_parse>` route with *integration_token* as ``account_token``, the document you send is not parsed. The ``document_id`` you will receive on success is a fake id you can use to test the rest of the flow.
+
+Since the document is not parsed when you call :ref:`/get_results <webservices/extract_api/invoice_get_results>`  you will always receive the same hard coded result. This result follows the response structure described in the documentation.
+
+In integration mode the route :ref:`/validate <webservices/extract_api/invoice_validate>` might return a ``status_code`` 12 to indicate that the format of your validation is incorrect. When using a real ``account_token``, the ``status_code`` will always be 0 even if the format is incorrect.
```

Comment:
We should probably change that behavior and return the 12 status code when the validation format is incorrect in non-integration mode.

Reply:
Agree! It will just add the validation but I think it is very fast so it should not be a problem

Reply:
I agree too :)

Reply:
You should remove the last sentence and adapt the first in that case.